applications/nrf_desktop/src/modules/ble_scan.c

```diff
@@ -202,12 +277,16 @@ static void discovery_completed(struct bt_gatt_dm *dm, void *context)
 
 static void discovery_service_not_found(struct bt_conn *conn, void *context)
 {
-	LOG_WRN("Cannot find service during the discovery");
+	LOG_ERR("Cannot find service during the discovery");
+	__ASSERT_NO_MSG(discovering_peer_conn == conn);
+	bt_conn_disconnect(conn, BT_HCI_ERR_REMOTE_USER_TERM_CONN);
```

Comment:
no unref?

Reply:
discovering_peer_conn  = NULL

Reply:
if this is handled from EM event please add a comment here

Reply:
It is handled from Event Manager (PEER_DISCONNECTED event). Added comment.